packages/backend-core/src/features/index.ts

```diff
@@ -53,38 +120,84 @@ function isFlagName(name: string): name is keyof Flags {
  * they will be accessed through this function as well.
  */
 export async function fetch(): Promise<Flags> {
-  const currentTenantId = context.getTenantId()
-  const flags = defaultFlags()
-
-  const split = (env.TENANT_FEATURE_FLAGS || "")
-    .split(",")
-    .map(x => x.split(":"))
-  for (const [tenantId, ...features] of split) {
-    if (!tenantId || (tenantId !== "*" && tenantId !== currentTenantId)) {
-      continue
-    }
+  return await tracer.trace("features.fetch", async span => {
+    const tags: Record<string, any> = {}
+
+    const flags = defaultFlags()
 
-    for (let feature of features) {
-      let value = true
-      if (feature.startsWith("!")) {
-        feature = feature.slice(1)
-        value = false
+    const currentTenantId = context.getTenantId()
+    const specificallySetFalse = new Set<string>()
+
+    const split = (env.TENANT_FEATURE_FLAGS || "")
+      .split(",")
+      .map(x => x.split(":"))
+    for (const [tenantId, ...features] of split) {
+      if (!tenantId || (tenantId !== "*" && tenantId !== currentTenantId)) {
+        continue
       }
 
-      if (!isFlagName(feature)) {
-        throw new Error(`Feature: ${feature} is not an allowed option`)
+      for (let feature of features) {
+        let value = true
+        if (feature.startsWith("!")) {
```

Comment:
Do we support negative flags? What is the use case of this?

Reply:
We do. I'm not sure, but it's a thing that existed before this PR and I'm maintaining the behaviour.

Reply:
Interesting 😅